### PR TITLE
added the functionality to send scheduled standup messages

### PR DESCRIPTION
**Pull request by ayushdevelop**
# Related Issue

- https://github.com/code-mancers/g-bot/issues/12

# Functionalities Added or Changed

- Added the functionality to send scheduled message for collecting standup.

# Additional Info

- added the package `node-cron`

### CODE REVIEW

File: server/src/initiate standup/initiateStandup.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+require("dotenv").config();
```
Comment (SujayPrabhu96) [inline server/src/initiate standup/initiateStandup.ts:1]: There should be white spaces in folder names.

File: server/src/scheduled standup/standupScheduler.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const scheduledStandup = require("../initiate standup/initiateStandup");
```
Comment (SujayPrabhu96) [inline server/src/scheduled standup/standupScheduler.ts:1]: same as above w.r.t folder names

File: server/src/scheduled standup/standupScheduler.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const cron = require("node-cron");
+
+/**
+ * This function is used to send a message
+ * to users reminding them for their standup
+ * submission
+ */
+
+const standupScheduler = (app: any) => {
+  cron.schedule("7 9 * * 1-5", function () {
```
Comment (SujayPrabhu96) [inline server/src/scheduled standup/standupScheduler.ts:11]: Add a comment for this, saying when it runs

Comment (SujayPrabhu96) [reply server/src/scheduled standup/standupScheduler.ts:11]: Let it be arrow function

File: server/src/initiate standup/initiateStandup.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  context,
+}: ScheduleStandupArguments): Promise<void> => {
+  const users: string[] | undefined = process.env.USERS?.split(" ");
+
+  users?.forEach(async (user: any) => {
+    try {
+      await client.chat.postMessage({
+        token: client.token,
+        // Channel to send message to
+        channel: `${user}`, //,
```
Comment (SujayPrabhu96) [inline server/src/initiate standup/initiateStandup.ts:19]:
```suggestion
        channel: user, //,
```

File: server/package.json [hunk 5]
```diff
@@ -13,7 +13,9 @@
     "@slack/bolt": "^3.12.1",
     "body-parser": "^1.20.0",
     "dotenv": "^16.0.2",
-    "express": "^4.18.1"
+    "express": "^4.18.1",
+    "node-cron": "^3.0.2",
+    "shelljs": "^0.8.5"
```
Comment (SujayPrabhu96) [inline server/package.json:18]: Whats the need to shell js ?

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

File: server/src/scheduled standup/standupScheduler.ts (hunk 2 above)
Comment (SujayPrabhu96) [reply server/src/scheduled standup/standupScheduler.ts:1]: Let the folder name be `/schedulers`

File: server/package.json (hunk 5 above)
Comment (ayushdevelop) [reply server/package.json:18]: oh no need of it, was a mistake will remove it

File: server/src/services/postMessage.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  token: string;
+  channel: string;
+};
+
+const postStandupMessage = async ({
+  token,
+  channel,
+  client,
+}: PostStandupMessageArgs) => {
+  console.log(standupMessageMarkup);
```
Comment (SujayPrabhu96) [inline server/src/services/postMessage.ts:14]: remove

File: server/src/services/postMessage.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+};
+
+const postStandupMessage = async ({
+  token,
+  channel,
+  client,
+}: PostStandupMessageArgs) => {
+  try {
+    await client.chat.postMessage({
+      token: token,
```
Comment (SujayPrabhu96) [inline server/src/services/postMessage.ts:16]:
```suggestion
      token,
```

File: server/src/services/postMessage.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      // Channel to send message to
+      channel: channel,
```
Comment (SujayPrabhu96) [inline server/src/services/postMessage.ts:18]:
```suggestion
      channel,
```

**Review by SujayPrabhu96 — CHANGES_REQUESTED** [review]

**Review by SujayPrabhu96 — APPROVED** [review]
LGTM

**Pull request merged**